packages/components/src/validated-form-controls/components/checkbox-control.tsx

```diff
@@ -0,0 +1,61 @@
+/**
+ * WordPress dependencies
+ */
+import { useMergeRefs } from '@wordpress/compose';
+import { forwardRef, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { ControlWithError } from '../control-with-error';
+import type { ValidatedControlProps } from './types';
+import CheckboxControl from '../../checkbox-control';
+import type { CheckboxControlProps } from '../../checkbox-control/types';
+
+type Value = CheckboxControlProps[ 'checked' ];
+
+const UnforwardedValidatedCheckboxControl = (
+	{
+		required,
+		customValidator,
+		onChange,
+		markWhenOptional,
+		...restProps
+	}: Omit< CheckboxControlProps, '__nextHasNoMarginBottom' > &
```

Comment:
`CheckboxControlProps` is actually just a subset of `React.ComponentProps< typeof CheckboxControl >`, and we need the latter set here. (Same with all the other components)

```suggestion
	}: Omit< React.ComponentProps< typeof CheckboxControl >, '__nextHasNoMarginBottom' > &
```

Reply:
Addressed with c571df83706b256094aae8e0a9bfec444d507e4e